### PR TITLE
Add support for linking OCPBUGS Jira Issues to tests and jobs

### PR DESCRIPTION
**Pull request by dgoodwin**
Adds support for linking jira issues to tests and jobs, similar to the old bugzilla code. Uses search.ci as before. This only scans jira issues in the new OCPBUGS project, OCPBUGSM and bugzilla remain unused. (as they would require maintaining two implementations)

Searches for tests by name, and jobs by name but in the previous job=[jobname]=all syntax to avoid partial matches. (which is actually a problem for tests in some rare cases)

### CODE REVIEW

**Comment by dgoodwin** [comment]
One remaining piece of work is to expose open_bugs in the jobs API and table, struggling with this a bit but wanted to get the bulk of the work pushed up to start review as this can land whenever.

File: pkg/api/job_analysis.go [hunk 1]
```diff
@@ -2,82 +2,46 @@ package api
...
+type JobAnalysisResult struct {
+	ByPeriod map[string]AnalysisResult `json:"by_period"`
 }
 
-func PrintJobAnalysisJSONFromDB(w http.ResponseWriter, req *http.Request, dbc *db.DB, release string) {
-	fil, err := filter.ExtractFilters(req)
-	if err != nil {
-		RespondWithJSON(http.StatusBadRequest, w, map[string]interface{}{"code": http.StatusBadRequest, "message": "Could not marshal query:" + err.Error()})
-		return
+func PrintJobAnalysisJSONFromDB(
```
Comment (dgoodwin) [inline pkg/api/job_analysis.go:27]: Undertook a refactor here to get at the parts of this process I wanted, namely to get to the job IDs, but also just to generally clean up how far we pass the http request and response writer.

**Comment by dgoodwin** [comment]
Alright I think I've fixed everything, both test and jobs apis now show open bugs count and support filtering. 
I am PTO until Tue so will leave this WIP, but I will start on review as soon as I'm back if anyone has time between now and then.

**Review by stbenjam — COMMENTED** [review]
Awesome! Just a couple of comments

File: sippy-ng/build/index.html [hunk 2]
```diff
@@ -1,10 +1 @@
-<!-- go:embed is unhappy with any empty directory so this services as a placeholder -->
```
[the remark this answers is not in the document]
Comment (stbenjam) [reply sippy-ng/build/index.html:1]: We're probably missing this from .gitignore

File: pkg/api/job_analysis.go [hunk 3]
```diff
@@ -2,82 +2,46 @@ package api
...
+	jobFilter *filter.Filter,
+	jobRunsFilter *filter.Filter,
+	start, boundary, end time.Time,
+	limit int,
+	sortField string,
+	sort apitype.Sort,
+	period string) (JobAnalysisResult, error) {
+	result := JobAnalysisResult{}
+
+	jobs, err2 := query.ListFilteredJobIDs(dbc, release, jobFilter,
```
[the remark this answers is not in the document]
Comment (dgoodwin) [reply pkg/api/job_analysis.go:39]: I think an artifact of my use of goland's extract method, will fix.

File: sippy-ng/build/index.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (dgoodwin) [reply sippy-ng/build/index.html:1]: Adding /sippy-ng/build/* to gitignore, hopefully ok to leave this in the diff, not sure how to get it out.

File: sippy-ng/src/jobs/JobTable.js [hunk 4]
```diff
@@ -80,6 +79,13 @@ export const getColumns = (config, openBugzillaDialog) => {
         </div>
       ),
     },
+    open_bugs: {
+      field: 'open_bugs',
+      headerName: 'Bugs',
+      type: 'number',
+      flex: 0.5,
+      renderCell: (params) => <div className="test-name">{params.value}</div>,
```
[the remark this answers is not in the document]
Comment (dgoodwin) [reply sippy-ng/src/jobs/JobTable.js:87]: Will do.

**Comment by dgoodwin** [comment]
Feedback addressed, thanks for review.

**Comment by stbenjam** [comment]
Let's land it and fix the build html as a follow-up

/lgtm

**Pull request merged**